Actions/ReadSecrets/ReadSecrets.ps1

```diff
@@ -22,151 +22,92 @@ try {
     }
 
     . (Join-Path -Path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)
-    Import-Module (Join-Path $PSScriptRoot ".\ReadSecretsHelper.psm1") -ArgumentList $gitHubSecrets
+    Import-Module (Join-Path $PSScriptRoot "ReadSecretsHelper.psm1")
+
+    $ghSecrets = $gitHubSecrets | ConvertFrom-Json
 
     $outSecrets = [ordered]@{}
-    $settings = $env:Settings | ConvertFrom-Json | ConvertTo-HashTable
-    $keyVaultCredentials = GetKeyVaultCredentials
-    $getAppDependencySecrets = $false
+    $keyVaultCredentials = GetKeyVaultCredentials -jsonStr $ghSecrets.AZURE_CREDENTIALS
     $getTokenForPush = $false
-    [System.Collections.ArrayList]$secretsCollection = @()
-    foreach($secret in ($getSecrets.Split(',') | Select-Object -Unique)) {
+    $secrets = @($ghSecrets.PSObject.Properties.Name)
+    for ($sno = 0; $sno -lt $secrets.Count; $sno++) {
+        $secret = $secrets[$sno]
+        $secretName = (get-item "ENV:S$sno").value
```

Comment:
That part shouldn't work if secret names are not env variables.

Reply:
Good catch - forgot about this one - we actually have to have the mapping in an ENV variable

Reply:
reverted back to use ENV - will run the tests to see if I got all